src/dataservice_publisher/resources/login.py

```diff
@@ -24,8 +24,10 @@ def post(self) -> Response:
             response.status_code = 401
             return response
 
-        username = request.json.get("username", None)
-        password = request.json.get("password", None)
+        data = request.get_json()
+        if data:
+            username = data.get("username", None)
+            password = data.get("password", None)
 
         if username != ADMIN_USERNAME or password != ADMIN_PASSWORD:
```

Comment:
Er det risiko for en `NameError` her? _username_ is not defined?

Reply:
Du tenker kanskje på `KeyError`? I så fall: nei. Ref https://docs.python.org/3/library/stdtypes.html?highlight=dict%20get#dict.get

Reply:
Ah, ser poenget ditt. Fikser.

Reply:
Det som er litt tricky (og implisitt), er at request.get_json() kaster 400 Bad Request error dersom det requesten ikkje har/har dårlige json-data. Dermed vil username og password alltid vere definert i praksis. Men eg legger til litt kode som gjer logikken klarare.
Ref https://flask.palletsprojects.com/en/2.2.x/api/?highlight=request%20json#flask.Request.get_json

Reply:
Fiksa i 9587ab78d2236580d060dccead034ada48edda11